File: frameworks/shared/java/src/main/java/com/velocitybench/healthcheck/HealthCheckConfig.java
package com.velocitybench.healthcheck;

/**
 * Health check manager configuration.
 */
public class HealthCheckConfig {
    private String serviceName;
    private String version;
    private String environment;
    private long startupDurationMs;

    public HealthCheckConfig() {
        this.serviceName = "velocitybench";
        this.version = "1.0.0";
        this.environment = "development";
        this.startupDurationMs = 30000; // 30 seconds
    }

    public HealthCheckConfig(String serviceName, String version, String environment, long startupDurationMs) {
        this.serviceName = serviceName;
        this.version = version;
        this.environment = environment;
        this.startupDurationMs = startupDurationMs;
    }

    // Getters and setters
    public String getServiceName() {
        return serviceName;
    }

    public void setServiceName(String serviceName) {
        this.serviceName = serviceName;
    }

    public String getVersion() {
        return version;
    }

    public void setVersion(String version) {
        this.version = version;
    }

    public String getEnvironment() {
        return environment;
    }

    public void setEnvironment(String environment) {
        this.environment = environment;
    }

    public long getStartupDurationMs() {
        return startupDurationMs;
    }

    public void setStartupDurationMs(long startupDurationMs) {
        this.startupDurationMs = startupDurationMs;
    }
}

File: frameworks/shared/csharp/VelocityBench.HealthCheck/HealthCheckConfig.cs
namespace VelocityBench.HealthCheck;

/// <summary>
/// Health check manager configuration.
/// </summary>
public class HealthCheckConfig
{
    public string ServiceName { get; set; } = "velocitybench";
    public string Version { get; set; } = "1.0.0";
    public string Environment { get; set; } = "development";
    public long StartupDurationMs { get; set; } = 30000; // 30 seconds

    public HealthCheckConfig()
    {
    }

    public HealthCheckConfig(string serviceName, string version, string environment, long startupDurationMs = 30000)
    {
        ServiceName = serviceName;
        Version = version;
        Environment = environment;
        StartupDurationMs = startupDurationMs;
    }
}
